modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/processors/OperationProcessor.java

```diff
@@ -70,8 +70,8 @@ public void processOperation(Operation operation) {
             for (String responseCode : responses.keySet()) {
                 ApiResponse response = responses.get(responseCode);
                 if(response != null) {
-                    //This part allows parser to put response inline without the resolveFully option set to true
-                    if (response.get$ref() != null) {
+                    //This part allows parser to put response inline within resolveResponses or ResolveFully
+                    if (response.get$ref() != null && cache != null && cache.getParseOptions() != null && (cache.getParseOptions().isResolveResponses() || cache.getParseOptions().isResolveFully())) {
```

Comment:
this shouldn't be triggered by `resolveFully` but only by the flag. It's true that we currently don't process `components/responses` in `ResolveFully` as we do for requestBodies, but this should be addressed in separate ticket and logic added to `ResolveFully` and not here